Add Software Development Contract - internal

ibrcko opened the pull request:
This document is designed to explain in what manner will software dev. dept. members contribute to a project.

All suggestions are very welcome, especially regarding each role expectations section! 

This still in development. I was experiencing a creative blockade, so every suggestion, complaint etc., is very well welcome 😃 

Relates: #37

ibrcko committed:
Add Software Development Contract - internal

This document is designed to explain in what manner will software dev. dept. members contribute to a project.

ibrcko committed:
Add Software Development Contract - internal

Cleanup after git hooks by adding soft. dev. contract to TOC

krukru commented:
Looks real nice! Please put this draft up on the dev weekly agenda 👍

ibrcko committed:
Generate TOC and add lock emoji before private document

renatoruk commented:
I've previously read the Croatian version and it's awesome. I think other teams and companies would benefit from this approach also. Could be a great candidate for a blog post 😎

Commit:
Update software-development-internal-contract.md

Replace client name with example in a hyperlink to example functional specification

napravicukod commented on playbooks/software-development-internal-contract.md:
```diff
@@ -0,0 +1,173 @@
+
+## Functional specification
+
+-   An official document without which the development shouldn’t be started
```
```suggestion
-   An official document as an absolute requirement for development to start
```

napravicukod commented on playbooks/software-development-internal-contract.md:
```diff
@@ -0,0 +1,173 @@
+
+-   An official document without which the development shouldn’t be started
+-   It provides an answer to:
+    -   Which software functionalities are we going to develop?
```
```suggestion
    -   Which software functionalities are to be developed?
```

napravicukod commented on playbooks/software-development-internal-contract.md:
```diff
@@ -0,0 +1,173 @@
+-   By using it:
+    -   Differences between client expectations, and our implementation can be greatly decreased
+    -   The chances of misunderstanding between people involved in a project can be decreased
+    -   There is a “single source of truth” to which all parties can rely upon if a misunderstanding occurs
```
```suggestion
    -   There is a “single source of truth” for reference in case of misunderstanding
```

napravicukod commented on playbooks/software-development-internal-contract.md:
```diff
@@ -0,0 +1,173 @@
+    -   Differences between client expectations, and our implementation can be greatly decreased
+    -   The chances of misunderstanding between people involved in a project can be decreased
+    -   There is a “single source of truth” to which all parties can rely upon if a misunderstanding occurs
+    -   A clear difference can be seen between “functionalities agreed upon" and "change requests"
```
```suggestion
    -   A change request is easier to define
```

napravicukod commented on playbooks/software-development-internal-contract.md:
```diff
@@ -0,0 +1,173 @@
+    -   There is a “single source of truth” to which all parties can rely upon if a misunderstanding occurs
+    -   A clear difference can be seen between “functionalities agreed upon" and "change requests"
+    -   Efficiency increases because we can predict potential problems from the very start of software development
+    -   A need for back-and-forth communication between a client and us is decreased
```
```suggestion
    -   A need for back-and-forth communication between the client and our side is decreased
```

napravicukod commented on playbooks/software-development-internal-contract.md:
```diff
@@ -0,0 +1,173 @@
+
+    -   Actors: client, user, tablet application
+    -   Requirements:
+        -   I, as a user, have a new client, and I want to store his information. This functionality is used in two
```
```suggestion
        -   I, as a user, have a new client, and I want to store this client's information. This functionality is used in two
```

napravicukod commented on playbooks/software-development-internal-contract.md:
```diff
@@ -0,0 +1,173 @@
+    -   Exclusively responsible for software delivery
+    -   Senior-level or Middle-level developers
+    -   Is in charge of task break down
+    -   _This role brings with it an increased level of responsibility for the software delivery in a predicted
```
```suggestion
    -   _This role comes with increased level of responsibility for the software delivery in a predicted
```

napravicukod commented on playbooks/software-development-internal-contract.md:
```diff
@@ -0,0 +1,173 @@
+    -   What can you expect from the person performing this role:
+        -   To take responsibility for software delivery
+        -   To be familiar with software business logic in detail
+        -   To know the answer to all of your questions about software requirements, business logic or technical
```
```suggestion
        -   To know answers to all of your questions about software requirements, business logic or technical
```

napravicukod commented on playbooks/software-development-internal-contract.md:
```diff
@@ -0,0 +1,173 @@
+-   Maker
+    -   All seniority levels
+    -   A person that is in charge of implementing features based on feature tasks and subtasks
+    -   Has a lot less responsibility for software delivery
```
```suggestion
    -   Delivery responsibility is mainly focused on tasks and features, depending on the maker seniority level
```

napravicukod commented on playbooks/software-development-internal-contract.md:
```diff
@@ -0,0 +1,173 @@
+    -   What can you expect from the person performing this role:
+        -   To do feature implementation based on the tasks and subtasks
+        -   To estimate feature implementation
+        -   To communicate its road-block, impediments or requests help
```
```suggestion
        -   To communicate roadblocks, impediments and ask for help
```

napravicukod commented on playbooks/software-development-internal-contract.md:
```diff
@@ -0,0 +1,173 @@
+        -   To communicate its road-block, impediments or requests help
+        -   To have a responsibility towards the Tech. Lead for feature implementation
+    -   What you can’t expect from the person performing this role:
+        -   Complete independence while doing its work
```
```suggestion
        -   Complete autonomy while working
```

napravicukod commented on playbooks/software-development-internal-contract.md:
```diff
@@ -0,0 +1,173 @@
+        -   To have a responsibility towards the Tech. Lead for feature implementation
+    -   What you can’t expect from the person performing this role:
+        -   Complete independence while doing its work
+        -   To take responsibility for software delivery
```
```suggestion
        -   To take the final responsibility for software delivery
```

Commit:
Update playbooks/software-development-internal-contract.md

Co-Authored-By: Rudolf Jurišić <napraviculom@gmail.com>

Commit:
Update playbooks/software-development-internal-contract.md

Co-Authored-By: Rudolf Jurišić <napraviculom@gmail.com>

Commit:
Update playbooks/software-development-internal-contract.md

Co-Authored-By: Rudolf Jurišić <napraviculom@gmail.com>

Commit:
Update playbooks/software-development-internal-contract.md

Co-Authored-By: Rudolf Jurišić <napraviculom@gmail.com>

Commit:
Update playbooks/software-development-internal-contract.md

Co-Authored-By: Rudolf Jurišić <napraviculom@gmail.com>

Commit:
Update playbooks/software-development-internal-contract.md

Co-Authored-By: Rudolf Jurišić <napraviculom@gmail.com>

Commit:
Update playbooks/software-development-internal-contract.md

Co-Authored-By: Rudolf Jurišić <napraviculom@gmail.com>

Commit:
Update playbooks/software-development-internal-contract.md

Co-Authored-By: Rudolf Jurišić <napraviculom@gmail.com>

Commit:
Update playbooks/software-development-internal-contract.md

Co-Authored-By: Rudolf Jurišić <napraviculom@gmail.com>

Commit:
Update playbooks/software-development-internal-contract.md

Co-Authored-By: Rudolf Jurišić <napraviculom@gmail.com>

Commit:
Update playbooks/software-development-internal-contract.md

Co-Authored-By: Rudolf Jurišić <napraviculom@gmail.com>

Commit:
Update playbooks/software-development-internal-contract.md

Co-Authored-By: Rudolf Jurišić <napraviculom@gmail.com>

marojeee commented:
Another important _process describer_, which should improve our day-to-day work.
Since I already gave my input in the Croatian version, I will just leave a thumbs up as a support for making this document and translating it  👍

krukru commented:
Ready to merge?

krukru commented:
> Ready to merge?

Guess not :D
